libbeat/esleg/eslegclient/bulkapi_mock_test.go

```diff
@@ -186,11 +174,89 @@ func TestEnforceParameters(t *testing.T) {
 		body = append(body, op)
 	}
 
-	params := map[string]string{
-		"refresh": "true",
+	tests := map[string]struct {
+		preconfigured map[string]string
+		reqParams     map[string]string
+		expected      map[string]string
+	}{
+		"Preconfigured parameters are applied to bulk requests": {
+			preconfigured: map[string]string{
+				"hello": "world",
+			},
+			expected: map[string]string{
+				"hello": "world",
+			},
+		},
+		"Preconfigured and local parameters are merged": {
+			preconfigured: map[string]string{
+				"hello": "world",
+			},
+			reqParams: map[string]string{
+				"foo": "bar",
+			},
+			expected: map[string]string{
+				"hello": "world",
+				"foo":   "bar",
+			},
+		},
+		"Local parameters only": {
+			reqParams: map[string]string{
+				"foo": "bar",
+			},
+			expected: map[string]string{
+				"foo": "bar",
+			},
+		},
+		"no parameters": {
+			expected: map[string]string{},
+		},
+		"Local overrides preconfigured parameters": {
+			preconfigured: map[string]string{
+				"foo": "world",
+			},
+			reqParams: map[string]string{
+				"foo": "bar",
+			},
+			expected: map[string]string{
+				"foo": "bar",
+			},
+		},
 	}
 
-	client.Bulk(context.Background(), index, "type1", params, body)
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			client, _ := NewConnection(ConnectionSettings{
+				Parameters: test.preconfigured,
+				URL:        "http://localhost",
+				Timeout:    0,
+			})
+
+			client.Encoder = NewJSONEncoder(nil, false)
+
+			var recParams url.Values
+			errShort := errors.New("shortcut")
+
+			client.HTTP = &reqInspector{
+				assert: func(req *http.Request) (*http.Response, error) {
+					recParams = req.URL.Query()
+					return nil, errShort
+				},
+			}
+
+			_, _, err := client.Bulk(context.Background(), index, "type1", test.reqParams, body)
+			require.Equal(t, errShort, err)
+			require.Equal(t, len(recParams), len(test.expected))
+
+			for k, v := range test.expected {
+				assert.Equal(t, recParams.Get(k), v)
+			}
```

Comment:
`assert.Equal(t, test.expected, recParams)` should actually work.

Reply:
+/- the  `url.Values` is a `map[string][]string`, which I find a bit weird, where every accessor only deals with key/value.. :

Reply:
Does it make sense to add another check that `recParams` does not include additional fields not found in `test.exptected`?

Reply:
We already do that by comparing the `len` of the map, so the ` len` all items need to match.
https://github.com/elastic/beats/pull/18393/files#diff-7bf553dedc14bb868bc139267101d16cR248